mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java

```diff
@@ -130,38 +121,26 @@ public CassandraMessageIdMapper(MailboxMapper mailboxMapper, CassandraMailboxDAO
                         .completableFuture());
     }
 
-    private CompletableFuture<Boolean> mailboxExists(MessageWithoutAttachment messageWithoutAttachment) {
-        CassandraId cassandraId = (CassandraId) messageWithoutAttachment.getMailboxId();
+    private CompletableFuture<Stream<SimpleMailboxMessage>> filterMessagesWIthExistingMailbox(Stream<SimpleMailboxMessage> stream) {
+        return FluentFutureStream.of(stream.map(this::mailboxExists))
+            .flatMap(m -> m)
```

Comment:
No, it's not Function<S, S> it's Function<S, STREAM<U>> with S being STREAM<U> thus identity don't apply (type checking issues)